test/demo.jl

```diff
@@ -105,3 +105,58 @@ end
     @test "μ" ∈ labels
     @test "σ" ∈ labels
 end
+
+# -- test ADTypes compatibility
+
+@testset "Demo with ADTyes" begin
+    # Define the target distribution using the `LogDensityProblem` interface
+    struct DemoProblem
+        dim::Int
+    end
+
+    LogDensityProblems.logdensity(p::DemoProblem, θ) = logpdf(MvNormal(zeros(p.dim), I), θ)
+    LogDensityProblems.dimension(p::DemoProblem) = p.dim
+    LogDensityProblems.capabilities(::Type{DemoProblem}) =
+        LogDensityProblems.LogDensityOrder{0}()
+
+    # Choose parameter dimensionality and initial parameter value
+    D = 10
+    initial_θ = rand(D)
+    ℓπ = DemoProblem(D)
+
+    # Set the number of samples to draw and warmup iterations
+    n_samples, n_adapts = 2_000, 1_000
+
+    # Define a Hamiltonian system
+    metric = DiagEuclideanMetric(D)
+    hamiltonian = Hamiltonian(metric, ℓπ, AutoForwardDiff())
```

Comment:
This is the only relevant part in the test, and it misses the case where the logdensity is provided as a LogDensityModel.